[Request] ReactiveCocoa Podspec update for 4.2 version

The title says it all! Thank you very much!

I have been in contact with @ashfurrow regarding this issue and he told me that the Podspec does not lint correctly (the linting errors are shown at the bottom). I have tried to investigate the issues and have figured out the following:

I think there are two main issues:
1. [ReactiveCocoa/UI] `'UIBarButtonItem+RACCommandSupport.h' file not found` is reported because the file is missing from the tvOS target. This one can easily be fixed by adding the `UIBarButtonItem` to the ss.tvos.source_files property.
2. [ReactiveCocoa/Core] `'ReactiveCocoa/ReactiveCocoa.h' file not found` is reported because the generated ReactiveCocoa.framework/Headers/ReactiveCocoa-Swift.h tries to `#import <ReactiveCocoa/ReactiveCocoa.h>` but the header file (ReactiveCocoa/ReactiveCocoa.h) is excluded from the ReactiveCocoa/Core-subspec.

If I include ReactiveCocoa/ReactiveCocoa.h file for the ReactiveCocoa/Core-subspec I meet another problem: The header file contains references to all the UI headers, which will fail with errors similar to `'UIBarButtonItem+RACCommandSupport.h' file not found`. This is only a problem when building the ReactiveCocoa/Core-subspec as a standalone module (without the UI).

To summarize:
- When building the ReactiveCocoa/Core-subspec the header file (ReactiveCocoa/ReactiveCocoa.h) **should not** include the UI headers.
- When building the ReactiveCocoa/UI-subspec the header file (ReactiveCocoa/ReactiveCocoa.h) **should** include the UI headers.

I am not sure how to differentiate between these two cases.

```
 -> ReactiveCocoa (4.2.0)
    - WARN  | [ReactiveCocoa/Core,ReactiveCocoa/UI] xcodebuild:  ReactiveCocoa/ReactiveCocoa/Objective-C/RACCompoundDisposable.m:85:12: warning: unused variable 'result' [-Wunused-variable]
    - WARN  | [ReactiveCocoa/Core,ReactiveCocoa/UI] xcodebuild:  ReactiveCocoa/ReactiveCocoa/Objective-C/RACCompoundDisposable.m:132:12: warning: unused variable 'result' [-Wunused-variable]
    - WARN  | [ReactiveCocoa/Core,ReactiveCocoa/UI] xcodebuild:  ReactiveCocoa/ReactiveCocoa/Objective-C/RACSerialDisposable.m:63:12: warning: unused variable 'result' [-Wunused-variable]
    - WARN  | [ReactiveCocoa/Core,ReactiveCocoa/UI] xcodebuild:  ReactiveCocoa/ReactiveCocoa/Objective-C/RACSerialDisposable.m:79:12: warning: unused variable 'result' [-Wunused-variable]
    - ERROR | [ReactiveCocoa/Core,ReactiveCocoa/UI] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | [ReactiveCocoa/UI] xcodebuild:  <module-includes>:1:9: note: in file included from <module-includes>:1:
    - NOTE  | [ReactiveCocoa/UI] xcodebuild:  Target Support Files/ReactiveCocoa/ReactiveCocoa-umbrella.h:75:9: note: in file included from Target Support Files/ReactiveCocoa/ReactiveCocoa-umbrella.h:75:
    - ERROR | [ReactiveCocoa/UI] xcodebuild:  ReactiveCocoa/ReactiveCocoa/ReactiveCocoa.h:67:10: error: 'UIBarButtonItem+RACCommandSupport.h' file not found
    - NOTE  | [ReactiveCocoa/Core,ReactiveCocoa/UI] xcodebuild:  <unknown>:0: error: could not build Objective-C module 'ReactiveCocoa'
    - NOTE  | [ReactiveCocoa/Core] xcodebuild:  <module-includes>:2:9: note: in file included from <module-includes>:2:
    - ERROR | [iOS] [ReactiveCocoa/Core] xcodebuild:  /Users/skohorn/Library/Developer/Xcode/DerivedData/App-gnksmemzphtghfdqgqlqkwopizxy/Build/Products/Release-iphonesimulator/ReactiveCocoa/ReactiveCocoa.framework/Headers/ReactiveCocoa-Swift.h:100:9: error: 'ReactiveCocoa/ReactiveCocoa.h' file not found
    - ERROR | [OSX] [ReactiveCocoa/Core] xcodebuild:  /Users/skohorn/Library/Developer/Xcode/DerivedData/App-gnksmemzphtghfdqgqlqkwopizxy/Build/Products/Release/ReactiveCocoa/ReactiveCocoa.framework/Headers/ReactiveCocoa-Swift.h:100:9: error: 'ReactiveCocoa/ReactiveCocoa.h' file not found
    - ERROR | [tvOS] [ReactiveCocoa/Core] xcodebuild:  /Users/skohorn/Library/Developer/Xcode/DerivedData/App-gnksmemzphtghfdqgqlqkwopizxy/Build/Products/Release-appletvsimulator/ReactiveCocoa/ReactiveCocoa.framework/Headers/ReactiveCocoa-Swift.h:100:9: error: 'ReactiveCocoa/ReactiveCocoa.h' file not found
    - ERROR | [watchOS] [ReactiveCocoa/Core] xcodebuild:  /Users/skohorn/Library/Developer/Xcode/DerivedData/App-gnksmemzphtghfdqgqlqkwopizxy/Build/Products/Release-watchsimulator/ReactiveCocoa/ReactiveCocoa.framework/Headers/ReactiveCocoa-Swift.h:100:9: error: 'ReactiveCocoa/ReactiveCocoa.h' file not found

Analyzed 1 podspec.

[!] The spec did not pass validation, due to 6 errors and 4 warnings.
```

Okay, I've hit a wall. I've trimmed down the podspec to basically nothing (just an iOS target) and when I lint I'm still getting the same error. Using CocoaPods 1.0.1, here's the output:

(Edit: I'm running this command in a directory where I've cloned the ReactiveCocoa repo, so using `pod lib lint` instead of `pod spec lint`.)

``` sh
> pod lib lint ReactiveCocoa.podspec --verbose --no-clean --allow-warnings

...

 -> ReactiveCocoa (4.2.1)
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | [iOS] xcodebuild:  <module-includes>:2:9: note: in file included from <module-includes>:2:
    - ERROR | [iOS] xcodebuild:  /Users/ash/Library/Developer/Xcode/DerivedData/App-fctegugttlqsjjegqurjwjwshruv/Build/Products/Release-iphonesimulator/ReactiveCocoa/ReactiveCocoa.framework/Headers/ReactiveCocoa-Swift.h:100:9: error: 'ReactiveCocoa/ReactiveCocoa.h' file not found
    - NOTE  | [iOS] xcodebuild:  <unknown>:0: error: could not build Objective-C module 'ReactiveCocoa'
```

(That output is trimmed.)

I feel like there's something easy here that I'm not getting. If I remember right, CocoaPods creates a MODULENAME/MODULENAME.h header file for Objective-C code, right? But when it's trying to reference that file in the Swift generated header, it breaks.

--- a/ReactiveCocoa.podspec
+++ b/ReactiveCocoa.podspec
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name = 'ReactiveCocoa'
-  s.version = '4.2.0'
+  s.version = '4.2.1'
   s.summary = 'A framework for composing and transforming streams of values.'
   s.description = <<-EOS
     ReactiveCocoa (RAC) is an Objective-C framework for Functional Reactive Programming.
@@ -15,56 +15,22 @@ Pod::Spec.new do |s|
   s.watchos.deployment_target = '2.0'
   s.source = {
     :git => 'https://github.com/ReactiveCocoa/ReactiveCocoa.git',
-    :tag => 'v4.2.0'
+    :tag => "v#{s.version}"
   }
   s.dependency 'Result', '~> 2.0'
   s.framework = 'Foundation'
-  s.default_subspec = 'UI'
-  s.prepare_command = "sed -i '' 's@<ReactiveCocoa/\\(.*\\)>@\"\\1\"@g' ReactiveCocoa/ReactiveCocoa.h"
+  s.private_header_files = [
+    '**/*Private.h',
+    '**/*EXTRuntimeExtensions.h',
+    '**/RACEmpty*.h'
+  ]
 
-  s.subspec 'no-arc' do |ss|
-    ss.source_files = 'ReactiveCocoa/Objective-C/RACObjCRuntime.{h,m}'
-    ss.requires_arc = false
-  end
+  s.source_files = 'ReactiveCocoa/**/*.{d,h,m,swift}'
 
-  s.subspec 'Core' do |ss|
-    ss.source_files = 'ReactiveCocoa/**/*.{d,h,m,swift}'
-    ss.exclude_files = [
-      '**/ReactiveCocoa.h',
-      'ReactiveCocoa/**/*{RACObjCRuntime,AppKit,NSControl,NSText,NSTable,UIActionSheet,UIAlertView,UIBarButtonItem,UIButton,UICollectionReusableView,UIControl,UIDatePicker,UIGestureRecognizer,UIImagePicker,UIRefreshControl,UISegmentedControl,UISlider,UIStepper,UISwitch,UITableViewCell,UITableViewHeaderFooterView,UIText,MK}*'
-    ]
-    ss.header_dir = 'ReactiveCocoa'
-    ss.private_header_files = [
-      '**/*Private.h',
-      '**/*EXTRuntimeExtensions.h',
-      '**/RACEmpty*.h'
-    ]
-    ss.dependency 'ReactiveCocoa/no-arc'
-
-    ss.watchos.exclude_files = '**/NSURLConnection*'
-    ss.watchos.pod_target_xcconfig = {'GCC_PREPROCESSOR_DEFINITIONS' => 'DTRACE_PROBES_DISABLED=1'}
-  end
-
-  s.subspec 'UI' do |ss|
-    ss.dependency 'ReactiveCocoa/Core'
-
-    ss.ios.source_files = [
-      '**/ReactiveCocoa.h',
-      'ReactiveCocoa/**/*{UIActionSheet,UIAlertView,UIBarButtonItem,UIButton,UICollectionReusableView,UIControl,UIDatePicker,UIGestureRecognizer,UIImagePicker,UIRefreshControl,UISegmentedControl,UISlider,UIStepper,UISwitch,UITableViewCell,UITableViewHeaderFooterView,UIText,MK}*'
-    ]
-    ss.ios.framework = 'UIKit'
-
-    ss.osx.source_files = [
-      '**/ReactiveCocoa.h',
-      'ReactiveCocoa/**/*{AppKit,NSControl,NSText,NSTable}*'
-    ]
-    ss.osx.framework = 'AppKit'
-
-    ss.tvos.source_files = [
-      '**/ReactiveCocoa.h',
-      'ReactiveCocoa/**/*{UIButton,UICollectionReusableView,UIControl,UIGestureRecognizer,UISegmentedControl,UITableViewCell,UITableViewHeaderFooterView,UIText}*'
-    ]
-
-    ss.watchos.source_files = '**/ReactiveCocoa.h'
-  end
+  # This is a little backwards, but it's basically easier to list the files we want to exlude
+  # the files we don't want than whitelist the ones we do. Could be room for improvement later.
+  s.ios.exclude_files = 'ReactiveCocoa/**/*{AppKit,NSControl,NSText,NSTable}*'
+  s.osx.exclude_files = 'ReactiveCocoa/**/*{UIActionSheet,UIAlertView,UIBarButtonItem,UIButton,UICollectionReusableView,UIControl,UIDatePicker,UIGestureRecognizer,UIImagePicker,UIRefreshControl,UISegmentedControl,UISlider,UIStepper,UISwitch,UITableViewCell,UITableViewHeaderFooterView,UIText,MK}*'
+  s.tvos.exclude_files = 'ReactiveCocoa/**/*{AppKit,NSControl,NSText,NSTable,UIActionSheet,UIAlertView,UIDatePicker,UIImagePicker,UIRefreshControl,UISlider,UIStepper,UISwitch,MK}*'
+  s.watchos.exclude_files = 'ReactiveCocoa/**/*{UIActionSheet,UIAlertView,UIBarButtonItem,UIButton,UICollectionReusableView,UIControl,UIDatePicker,UIGestureRecognizer,UIImagePicker,UIRefreshControl,UISegmentedControl,UISlider,UIStepper,UISwitch,UITableViewCell,UITableViewHeaderFooterView,UIText,MK,AppKit,NSControl,NSText,NSTable,NSURLConnection}*'
 end

